terminate called after throwing an instance of 'std::bad_weak_ptr'
  what():  bad_weak_ptr
bash: line 7:  4264 Aborted                 (core dumped) ./a.out
